Competition2017/src/competition/subsystems/shooter/SideShooterSubsystem.java

```diff
@@ -0,0 +1,166 @@
+package competition.subsystems.shooter;
+
+import org.apache.log4j.Logger;
+
+import com.ctre.CANTalon.FeedbackDevice;
+import com.ctre.CANTalon.TalonControlMode;
+
+import edu.wpi.first.wpilibj.Timer;
+import xbot.common.command.BaseSubsystem;
+import xbot.common.controls.actuators.XCANTalon;
+import xbot.common.injection.wpi_factories.WPIFactory;
+import xbot.common.properties.BooleanProperty;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+public class SideShooterSubsystem extends BaseSubsystem {
+    
+    private static Logger log = Logger.getLogger(ShooterSubsystem.class);
+    
+    private final XCANTalon masterMotor;
+    public final String side;
+    
+    // output telemetry properties
+    private final DoubleProperty shooterCurrentSpeed;
+    private final DoubleProperty shooterTargetSpeed;
+    private final DoubleProperty shooterOutputPower;
+    private final DoubleProperty shooterTalonError;
+    private final BooleanProperty atSpeedProp;
+    private final BooleanProperty enableShooterLogging;
+    
+    private final DoubleProperty p;
+    private final DoubleProperty i;
+    private final DoubleProperty d;
+    private final DoubleProperty f;
+    
+    private final DoubleProperty shooterSpeedThresh;
+    
+   
+    public SideShooterSubsystem(int motor, String side,WPIFactory factory, XPropertyManager propManager){
+        log.info("Creating" + side + " Shooter");
+        this.side = side;
+        
+        p = propManager.createPersistentProperty(side + "Shooter wheel P", 0);
```

Comment:
Since these values are being set into the CANTalon, it doesn't seem appropriate to use the PIDManager - especially since there's no way today to p/i/d values out of it.

This does suggest that we should build a PIDManagerLite that is just a useful container though!